xbmc/cores/IPlayer.h

```diff
@@ -208,6 +208,8 @@ class IPlayer
    */
   virtual void Render(bool clear, uint32_t alpha = 255, bool gui = true) {};
   virtual void FlushRenderer() {};
+  virtual void PreInitRenderer() {};
```

Comment:
pulling implementation details up here is an architectural disaster. not only that CApp should not even know that there is something like a renderer (and other players have none), exposing this is completely unsecure. calling this method at some other place/time will just crash the app.
sorry to say, this is an ugly hack.

Reply:
This is the original commit:
https://github.com/xbmc/xbmc/commit/7cd6b19131c54d5f11456ce284d066a3db124daf

I wonder how this was expected to work?
It sends a signal that is never handled, thus running always in a timeout when not called from the mainthread.
Then it doesn't return at all, tries to run the same code in the wrong threadcontext and deadlock or crash.
Why adding a message that is never handled?
For me it looked like handling the message in the mainthread was the missing part...